### PR TITLE
fix standalone viewer build via build-app adjustments

### LINKED ISSUE #767: Viewer assets return 404
**Describe the bug**
Fonts and svgs don't load on https://googlechrome.github.io/lighthouse-ci/viewer/

**To Reproduce**
1. Go to https://googlechrome.github.io/lighthouse-ci/viewer/

**Expected behavior**
Assets load as usual.

**Logs/Screenshots**
image

### PR DESCRIPTION
**Pull request by paulirish**
https://googlechrome.github.io/lighthouse-ci/viewer/ is busted
fixes #767

but if you build locally and run your server from inside /packages/viewer/dist its all good. the problem is some root-absolute urls... eg `<script src="/script-needed-from-origin-root.js">`

anyhow. This appears to be the sleekest solution that keeps both viewer and the server app looking good with no broken references.

i found a lot of weird behavior regarding `@chialab/esbuild-plugin-html'` and `entryNames`, `assetNames` and `chunkNames`. really weird. [filed this one](https://github.com/chialab/rna/issues/49).
but.. ultimately this new usage of that module feels like an improvement over the current status quo. 

<details>
<summary>some development notes</summary>

of course the root problem is plenty of 404s.

this is the primary cmd i used to iterate ..

```sh
# from viewer folder
find ../../scripts/build-app.js ./src/ui | entr bash -c "rm -rf ../viewer/dist/*; yarn build; tree dist"
```

</details>

### CODE REVIEW

File: scripts/build-app.js [hunk 1]
```diff
@@ -25,61 +25,31 @@ if (!['build', 'watch'].includes(command)) {
...
-  const stylesheets = Object.keys(result.metafile.outputs).filter(o => o.endsWith('.css'));
-  const stylesheet = stylesheets[0];
-  if (stylesheets.length === 0) return;
-  if (stylesheets.length > 1) throw new Error('expected at most one generated stylesheet');
-
-  const href = publicPath === '/' ?
-    path.relative(outdir, stylesheet) :
-    `/app/${path.relative(outdir, stylesheet)}`;
-
-  const htmls = Object.keys(result.metafile.outputs).filter(o => o.endsWith('.html'));
```
Comment (connorjclark) [inline scripts/build-app.js:40]: nice job untangling that mess

**Review by connorjclark — APPROVED** [review]
thanks for taking care of this

**Comment by paulirish** [comment]
@connorjclark thx.

these CI tests are just so flaky we give it a few attempts and then just merge anyway?

**Comment by paulirish** [comment]
i deployed https://googlechrome.github.io/lighthouse-ci/viewer/ and its lookin good

**Pull request merged**